assets/js/blocks/featured-product/block.js

```diff
@@ -224,7 +270,12 @@ class FeaturedProduct extends Component {
 			showDesc,
 			showPrice,
 		} = attributes;
-		const { loaded, product } = this.state;
+
+		// If there was an API error, render it.
+		if ( isObject( error ) ) {
```

Comment:
I think there is no need to check that `error` is an object, I guess any value of `error` that evaluates to true should trigger this check to be true.
```suggestion
		if ( error ) {
```